modules/_labs/select/react/lib/SelectBase.tsx

```diff
@@ -352,6 +368,11 @@ export default class SelectBase extends React.Component<SelectBaseProps> {
         value: option.value,
         ...(onOptionSelection
           ? {
+              // Prevent click from propagating (and, say, dismissing
+              // a modal containing the Select)
+              onClick: (event: React.MouseEvent) => {
+                event.stopPropagation();
+              },
               // mouseDown provides a slightly better UX than click
               // since visual feedback of selected option is more
               // immediate
```

Comment:
Not really part of this PR, but I did notice this while testing. Right-clicking selects a menu item. Or more accurately, right-mouse down selects an item. Why did we decide to do mouse down instead of a click? A native select waits for a click. If you press your mouse down, but don't release, the item is selected and the menu closes. It is surprising.

At minimum, we should not select on any other mouse button than the left one. At maximum, we should consider selection on click.

Reply:
Option selection will be modified to trigger on click instead of mousedown in #672.

Previously, I had set it to mousedown to avoid the brief moment when two options share the same blue background color (the just-clicked-on option and the assistive-focused option), but this is an extremely minor and subjective visual nit. I agree consistency with the native select should win out here.